src/vm.c

```diff
@@ -790,7 +790,8 @@ v7_val_t v7_create_string(struct v7 *v7, const char *p, size_t len, int own) {
     embed_string(m, m->len, (char *) &p, sizeof(p));
   }
 
-  return v7_pointer_to_value((void *) offset) | tag;
+  /* NOTE(lsm): don't use v7_pointer_to_value, 32-bit ptrs will truncate */
+  return (offset & ~V7_TAG_MASK) | tag;
```

Comment:
the approach with unions in go/typedef-val_t will make this kind of stuff easier to write and maintain.